### PR TITLE
Add remote read external label handling

### PR DESCRIPTION
**Pull request by juliusv**
This implements rule 1 and 2 from
https://docs.google.com/document/d/188YauRgfF0J4CYMigLsVNN34V_kUwKnApBs2dQMfBbs/edit

### CODE REVIEW

**Comment by juliusv** [comment]
@brian-brazil

File: storage/remote/read_test.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+			inMatchers: metric.LabelMatchers{
+				mustNewLabelMatcher(metric.Equal, "foo", "bar"),
+			},
+			outMatchers: metric.LabelMatchers{
+				mustNewLabelMatcher(metric.Equal, "foo", "bar"),
+			},
+			added: model.LabelSet{},
+		},
+		{
+			el: model.LabelSet{"biz": "buz", "baz": "boz"},
```
Comment (grobie) [inline storage/remote/read_test.go]: Hm, would you mind using some more descriptive examples here?

Comment (juliusv) [inline storage/remote/read_test.go]: How about now?

Comment (grobie) [inline storage/remote/read_test.go]: :+1:

File: storage/remote/read.go [hunk 2]
```diff
@@ -62,22 +64,68 @@ func (r *Reader) Queriers() []local.Querier {
...
+	res, err := q.client.Read(ctx, from, through, m)
+	if err != nil {
+		return nil, err
+	}
+
+	removeExternalLabels(res, added)
+	return res, err
+}
+
+func (q *querier) addExternalLabels(matchers metric.LabelMatchers) (metric.LabelMatchers, model.LabelSet) {
```
Comment (brian-brazil) [inline storage/remote/read.go]: This needs a comment so you can figure out what's going on.

File: storage/remote/read.go [hunk 3]
```diff
@@ -62,22 +64,68 @@ func (r *Reader) Queriers() []local.Querier {
...
+		if err != nil {
+			panic(err)
+		}
+		matchers = append(matchers, m)
+	}
+
+	return matchers, el
+}
+
+func removeExternalLabels(m model.Matrix, added model.LabelSet) {
```
Comment (brian-brazil) [inline storage/remote/read.go]: removeAddedLabels would be more accurate

**Comment by juliusv** [comment]
@brian-brazil Both comments addressed.

File: storage/remote/read.go [hunk 4]
```diff
@@ -120,7 +126,7 @@ func (q *querier) addExternalLabels(matchers metric.LabelMatchers) (metric.Label
 	return matchers, el
 }
 
-func removeExternalLabels(m model.Matrix, added model.LabelSet) {
+func removeLabels(m model.Matrix, added model.LabelSet) {
```
Comment (grobie) [inline storage/remote/read.go]: If you want to go down the generic naming route, I'd rename `added` to `labels` as well.

Comment (juliusv) [inline storage/remote/read.go]: Done.

**Comment by brian-brazil** [comment]
:+1:

**Pull request merged**